### ISSUE TITLE
Virtual machine: Removing security group attached to network

### ISSUE THREAD

**Issue by maartendq** [OpenNebula/terraform-provider-opennebula#387]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform: 1.3.3
Opennebula:  5.12.8
terraform-provider-opennebula: 1.1.0

### Affected Resource(s)

- opennebula_virtual_machine

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    opennebula = {
      source = "OpenNebula/opennebula"
      version = "1.1.0"
    }
  }
}

resource "opennebula_virtual_network" "network" {
   name            = "test"
   vlan_id         = "40"
   security_groups = ["0", "10"]
   network_mask    = "255.255.255.0"
   gateway         = "10.0.0.254"
   type            = "802.1"
   physical_device = "bond0"

resource "opennebula_virtual_machine" "vm" {
  name        = "testvm"
  description = "test"
  cpu         = 1
  vcpu        = 1
  memory      = 1
  group       = "oneadmin"
  permissions = "660"

  disk {
    image_id = 5
    size     = 4096
    target   = "vda"
    driver   = "qcow2"
  }

  nic {
    model = "virtio"
    ip = "10.0.0.2"
    network_id = "10"
    security_groups = "["11"]"
}
```

### Debug Output

```
  # module.one_vms["testvm"].opennebula_virtual_machine.vm will be updated in-place
  ~ resource "opennebula_virtual_machine" "vm" {
        id             = "1636"
        name           = "testvm"

      ~ nic {
          ~ security_groups          = [
              - 10,
                11,
                # (1 unchanged element hidden)
            ]
            # (10 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }
```

### Behavior

Note: We are migrating from terraform v0.15.5 and opennebula provider 0.4.0 to the versions described above. 
Previously we did not have the security group, that's attached to the nework, in the VM's NIC definition (I think it neglected this and it got attached anyhow by opennebula)
Terraform plan/apply is now trying to remove that security group. Is this intended behaviour? Should we always define security groups, that are attached to the virtual network, in the VM's NIC under security_groups too?

Furthermore, testing the actual apply results in the following:

```
╷
│ Error: Failed to update NIC
│
│   with module.one_vms["testvm"].opennebula_virtual_machine.vm,
│   on .terraform/modules/one_vms/vm.tf line 1, in resource "opennebula_virtual_machine" "vm":
│    1: resource "opennebula_virtual_machine" "vm" {
│
│ virtual machine (ID: 1636): vm nic attach: network 23: can't find the nic
```

**Comment by treywelsh** [comment]
This make me think to https://github.com/OpenNebula/terraform-provider-opennebula/issues/342#issuecomment-1293434715
reported for the v0.5.2 release of the provider and fixed from the 1.0.2 release.

**Comment by treywelsh** [comment]
Sorry small mistake, do you reproduce this problem from the 1.0.2 release of the provider ?
